app/controllers/graders_controller.rb

```diff
@@ -15,10 +15,30 @@ class GradersController < ApplicationController
   def index
     @assignment = Assignment.find(params[:assignment_id])
 
+    # We will complement the data hash returned by the Assignment.current_grader_data method by adding 'groups.members'
```

Comment:
This code has the right idea, but I have a few comments for improving it:

1. You shouldn't add additional code in the controller method; instead, add code to the `current_grader_data` method instead.
2. Right now the loop (`data[:groups].each`) has an **N + 1 query problem** (I recommend looking up what that means). Instead of doing this, a better approach is to make a single database query using `joins` that combines the `grouping` and `roles` (as student members) tables together. After doing so you can then group the results by `grouping_id` and then merge with the existing grouping data.

As a tip, I strongly recommend using the Rails console to experiment with making more complex active record queries, since you'll likely need to iterate a bunch to get this right.